### PR TITLE
DNS manage: Add notice when not using WPCOM nameservers

### PR DESCRIPTION
**Pull request by lupus2k**
Related to https://github.com/Automattic/dotcom-forge/issues/3485

## Proposed Changes

We want to add a notice on the DNS page when the domain nameservers are not managed by WPCOM as we did on Domain Redirect card when editing a domain.

| Domain Redirect | DNS |
| --- | --- |
| Image | image |

## Testing Instructions

* Disable WPCOM nameservers on a domain
* Go to edit DNS records
* You should see the notice
* The CTA should link to the edit domain page with the nameservers card open.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by candy02058912 — APPROVED** [review]
LGTM 👍 Notice renders!
renders

File: client/my-sites/domains/domain-management/dns/dns-records.jsx [hunk 1]
```diff
@@ -104,6 +108,56 @@ class DnsRecords extends Component {
...
+			<div className="dns-records-notice">
+				<Icon
+					icon={ info }
+					size={ 18 }
+					className="dns-records-notice__icon gridicon"
+					viewBox="2 2 20 20"
+				/>
+				<div className="dns-records-notice__message">
+					{ translate(
+						'Domain redirection requires using WordPress.com nameservers. {{a}}Update your nameservers now{{/a}}.',
```
Comment (candy02058912) [inline client/my-sites/domains/domain-management/dns/dns-records.jsx]: Rereading this, I wonder if this copy reflects well on the DNS page as they are setting the DNS settings and not domain redirection 🤔

Comment (lupus2k) [reply client/my-sites/domains/domain-management/dns/dns-records.jsx]: Thanks for noticing this. I use the Domain Transfer here by mistake 🤦

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8717530

Thank you @lupus2k for including a screenshot in the description! This is really helpful for our translators.

**Review by paulopmt1 — APPROVED** [review]
Working as expected, nice work @lupus2k!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8717530

Thank you @lupus2k for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8717530).

**Pull request merged**